packages/flutter/test/widgets/spell_check_test.dart

```diff
@@ -0,0 +1,318 @@
+    ]);
+
+    expectedTextSpanTree = TextSpan(children: <TextSpan>[
+      const TextSpan(text: 'Hello, '),
+      TextSpan(style: misspelledTextStyle, text: 'wrold'),
+      const TextSpan(text: '! '),
+      TextSpan(style: composingStyle, text: 'Hey')
+    ]);
+    textSpanTree = defaultSpellCheckSuggestionsHandler
+        .buildTextSpanWithSpellCheckSuggestions(
+            value, composingRegionOutOfRange, null, spellCheckResults);
+
+    expect(textSpanTree, equals(expectedTextSpanTree));
+  });
+
+  testWidgets(
+      'buildTextSpanWithSpellCheckSuggestions, composing region and misspelled words overlap example',
+      (WidgetTester tester) async {
+    text = 'Right worng worng right';
+    value = TextEditingValue(
+        text: text, composing: const TextRange(start: 12, end: 17));
+    composingRegionOutOfRange = false;
+    spellCheckResults =
+        SpellCheckResults(text, const <SuggestionSpan>[
+      SuggestionSpan(
+          TextRange(start: 6, end: 11), <String>['wrong', 'worn', 'wrung']),
+      SuggestionSpan(
+          TextRange(start: 12, end: 17), <String>['wrong', 'worn', 'wrung'])
+    ]);
+
+    expectedTextSpanTree = TextSpan(children: <TextSpan>[
+      const TextSpan(text: 'Right '),
+      TextSpan(style: misspelledTextStyle, text: 'worng'),
+      const TextSpan(text: ' '),
+      TextSpan(style: composingStyle, text: 'worng'),
+      const TextSpan(text: ' right'),
+    ]);
+    textSpanTree = defaultSpellCheckSuggestionsHandler
+        .buildTextSpanWithSpellCheckSuggestions(
+            value, composingRegionOutOfRange, null, spellCheckResults);
+
+    expect(textSpanTree, equals(expectedTextSpanTree));
+  });
+
+  testWidgets(
+      'buildTextSpanWithSpellCheckSuggestions, consecutive misspelled words example',
+      (WidgetTester tester) async {
+    text = 'Right worng worng right';
+    value = TextEditingValue(text: text);
+    composingRegionOutOfRange = true;
+    spellCheckResults =
+        SpellCheckResults(text, const <SuggestionSpan>[
+      SuggestionSpan(
+          TextRange(start: 6, end: 11), <String>['wrong', 'worn', 'wrung']),
+      SuggestionSpan(
+          TextRange(start: 12, end: 17), <String>['wrong', 'worn', 'wrung'])
+    ]);
+
+    expectedTextSpanTree = TextSpan(children: <TextSpan>[
+      const TextSpan(text: 'Right '),
+      TextSpan(style: misspelledTextStyle, text: 'worng'),
+      const TextSpan(text: ' '),
+      TextSpan(style: misspelledTextStyle, text: 'worng'),
+      const TextSpan(text: ' right'),
+    ]);
+    textSpanTree = defaultSpellCheckSuggestionsHandler
+        .buildTextSpanWithSpellCheckSuggestions(
+            value, composingRegionOutOfRange, null, spellCheckResults);
+
+    expect(textSpanTree, equals(expectedTextSpanTree));
+  });
+
+  testWidgets(
+      'buildTextSpanWithSpellCheckSuggestions corrects results when they lag, results text shorter than actual example',
```

Comment:
So these tests are testing the situation where `resultsText` is typed in and sent to be spell checked, then before the results come back the user types more text so the field contains `text`, and then `text` gets updated with the out-of-date spell check results?